clients/src/main/java/org/apache/kafka/clients/producer/internals/TransactionManager.java

```diff
@@ -289,6 +295,14 @@ synchronized void transitionToFatalError(RuntimeException exception) {
         transitionTo(State.FATAL_ERROR, exception);
     }
 
+    synchronized boolean isPartitionAdded(TopicPartition partition) {
```

Comment:
Perhaps add a '// visible for testing' comment to this and the next method? They really only should be called from tests.